### PR TITLE
Publish installers to dotnetcli / dotnetclichecksums for the dev chan…

### PR DESCRIPTION
**Pull request by riarenas**
Changes to enable publishing installers to dotnetcli and checksums to dotnetclichecksums for #3607

I have only made the changes for the public channels.

 I'm still testing to see if we can use proxies, or if the private version of this should wait until https://github.com/dotnet/arcade/issues/3684

I'll provide a newer test build when I hack my stuff around the dotnet-install stuff, but wanted to see if this is the path we want to use.

This also cannot be merged until we have a new arcade SDK that includes the changes in https://github.com/dotnet/arcade/pull/3696

I'm also thinking of adding a parameter to opt out of this publishing, for repos that already implemented something else for this.

### CODE REVIEW

File: eng/common/templates/post-build/common-variables.yml [hunk 1]
```diff
@@ -11,7 +12,7 @@ variables:
 
   # .NET Tools - Validation
   - name: PublicValidationRelease_30_Channel_Id
-    value: 9
+    value: 260
```
Comment (riarenas) [inline eng/common/templates/post-build/common-variables.yml]: Oops, I left this from my testing. Will remove

File: src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SetupTargetFeeds.proj [hunk 2]
```diff
@@ -116,13 +116,25 @@
...
 
+
     <!-- Configuration for public non-stable feeds -->
     <ItemGroup Condition="'@(TargetFeedConfig)' == ''">
       <TargetFeedConfig
         Include="NetCore;OSX;Deb;Rpm;Node;BinaryLayout;Installer;Checksum;Maven;VSIX"
         TargetURL="$(TargetStaticFeed)"
         Type="AzureStorageFeed"
         Token="$(AzureStorageTargetFeedPAT)" />
+      <TargetFeedConfig Condition="'$(InstallersAzureAccountKey)' != '' and '$(InstallersAzureAccountKey)' != ''"
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SetupTargetFeeds.proj]: Comparing the same variable

Comment (riarenas) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SetupTargetFeeds.proj]: derp

File: src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SetupTargetFeeds.proj [hunk 3]
```diff
@@ -116,13 +116,25 @@
       <TargetStaticFeed Condition="'$(TargetStaticFeed)' == ''">https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json</TargetStaticFeed>
     </PropertyGroup>
 
+
```
Comment (riarenas) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SetupTargetFeeds.proj]: I'll remove this extra line

**Comment by riarenas** [comment]
New test build with all the latest changes and publishing to a test container in the storage accounts:

https://dev.azure.com/dnceng/internal/_build/results?buildId=315910

Relevant logs:
```
  Artifact with category 'NetCore' should go to AzureStorageFeed -> 'https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json'
  Artifact with category 'OSX' should go to AzureStorageFeed -> 'https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json'
  Artifact with category 'Deb' should go to AzureStorageFeed -> 'https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json'
  Artifact with category 'Rpm' should go to AzureStorageFeed -> 'https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json'
  Artifact with category 'Node' should go to AzureStorageFeed -> 'https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json'
  Artifact with category 'BinaryLayout' should go to AzureStorageFeed -> 'https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json'
  Artifact with category 'Installer' should go to AzureStorageFeed -> 'https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json'
  Artifact with category 'Checksum' should go to AzureStorageFeed -> 'https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json'
  Artifact with category 'Maven' should go to AzureStorageFeed -> 'https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json'
  Artifact with category 'VSIX' should go to AzureStorageFeed -> 'https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json'
  Artifact with category 'Deb' should go to AzureStorageFeed -> 'https://dotnetcli.blob.core.windows.net/riarenas/index.json'
  Artifact with category 'Rpm' should go to AzureStorageFeed -> 'https://dotnetcli.blob.core.windows.net/riarenas/index.json'
  Artifact with category 'Node' should go to AzureStorageFeed -> 'https://dotnetcli.blob.core.windows.net/riarenas/index.json'
  Artifact with category 'BinaryLayout' should go to AzureStorageFeed -> 'https://dotnetcli.blob.core.windows.net/riarenas/index.json'
  Artifact with category 'Installer' should go to AzureStorageFeed -> 'https://dotnetcli.blob.core.windows.net/riarenas/index.json'
  Artifact with category 'Maven' should go to AzureStorageFeed -> 'https://dotnetcli.blob.core.windows.net/riarenas/index.json'
  Artifact with category 'VSIX' should go to AzureStorageFeed -> 'https://dotnetcli.blob.core.windows.net/riarenas/index.json'
  Artifact with category 'Checksum' should go to AzureStorageFeed -> 'https://dotnetclichecksums.blob.core.windows.net/riarenas/index.json'
```

Container in dotnetcli:
image

Container in dotnetclichecksums:
image


@mmitche and @JohnTortugo this is ready for review again.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SetupTargetFeeds.proj [hunk 4]
```diff
@@ -123,6 +136,16 @@
         TargetURL="$(TargetStaticFeed)"
         Type="AzureStorageFeed"
         Token="$(AzureStorageTargetFeedPAT)" />
+      <TargetFeedConfig Condition="'$(PublishInstallersAndChecksums)' == 'true'"
+        Include="Deb;Rpm;Node;BinaryLayout;Installer;Maven;VSIX"
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SetupTargetFeeds.proj]: This binary layout thing (.zip files, right?) is also an installer?

Comment (mmitche) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SetupTargetFeeds.proj]: Yeah, zips, msis, etc. should all be on this layout.  Does this handle that @riarenas

Comment (riarenas) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SetupTargetFeeds.proj]: Zip is handled by BinaryLayout, MSI is handled by Installer category

File: src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SetupTargetFeeds.proj [hunk 5]
```diff
@@ -43,6 +48,14 @@
       Condition="('$(AzureStorageAccountName)' == '' or '$(AzureStorageAccountKey)' == '') and '$(IsStableBuild)' == 'true'"
       Text="Parameters 'AzureStorageAccountName/Key' are empty." />
 
+    <Error
```
Comment (mmitche) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SetupTargetFeeds.proj]: Does this handle stabilized builds? Publishing to a location that will not overwrite existing binaries?

Comment (riarenas) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SetupTargetFeeds.proj]: For stabilized binaries we already create a new container / feed on every build and publish everything there.

Comment (mmitche) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SetupTargetFeeds.proj]: Okay, so the installers make it there too.

Comment (riarenas) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SdkTasks/SetupTargetFeeds.proj]: Yep

**Pull request merged**